extensions/amp-wistia-player/0.1/amp-wistia-player.js

```diff
@@ -76,9 +93,23 @@ class AmpWistiaPlayer extends AMP.BaseElement {
     iframe.src = '//fast.wistia.net/embed/iframe/' + encodeURIComponent(
         mediaId);
     this.applyFillContent(iframe);
-    this.element.appendChild(iframe);
     this.iframe_ = iframe;
-    return this.loadPromise(iframe);
+
+    this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleWistiaMessages_.bind(this)
+    );
+
+    this.element.appendChild(this.iframe_);
+
+    const loaded = this.loadPromise(this.iframe_).then(() => {
+      // Tell Wistia Player we want to receive messages
+      this.listenToFrame_();
+      this.element.dispatchCustomEvent(VideoEvents.LOAD);
+    });
+    this.playerReadyResolver_(loaded);
```

Comment:
It looks like this is being resolved when the iframe loads instead of when the player reports "ready." Am I reading it correctly? That might be fine functionally, but if I did read it right, I would rename it to avoid confusion. Maybe `playerLoadedPromise_` and `playerLoadedResolver_`?

Reply:
I've been trying to keep things consistent with https://github.com/ampproject/amphtml/blob/master/extensions/amp-youtube/0.1/amp-youtube.js

Reply:
Gotcha. I think this is somewhere we should depart from their language though--we have no obligation to have the same API as youtube. I think, for our maintenance, it makes the most sense to stay consistent with the rest of our code base.

Reply:
Will do